### DISCUSSION TITLE
FCPXML export/import problem with m4a,mp4 in DF

### DISCUSSION THREAD

**Discussion by tangierc** [CommandPost/FCPCafe#144] [Final Cut Pro Questions] [answered]
Hey folks. Came across a unique issue yesterday prepping my timeline for ProTools (via X2Pro).

The XML exported just fine, but couldn't be parsed by X2Pro. Coincidentally the same XML FCP 10.6.5 exported (XML 1.9 an 1.10 versions) couldn't be imported back into FCP without error.

Thanks to the help of another FCP user, it was determined that all of the problems came from mostly m4a clips and one mp4 in drop frame tc format. Technically FCP does not support that container according to its supported media formats page.

Example problem file in XML (AnxiousCut5Locked.fcpxml:33047:  clip lane="-3" offset="217896679/4000s" name="Maybe won't be bad.m4a" duration="11011/8000s" format="r18" tcFormat="DF"

I've had no issues importing and playing these clips on my timeline.  I only know that there is a problem with them is  X2Pro and FCP being unable to parse the FCP-created XML. Isn't odd that FCP didn't / wouldn't identify these issues when exporting the XML?

Welcome any suggestions.  The TC format column in the FCP browser doesn't reveal any DF or NDF info.

**Comment by latenitefilms** [comment]
Unfortunately Final Cut Pro doesn't validate their own FCPXMLs on export - so you can actually get FCPX to export invalid FCPXMLs.

I'd suggest sending feedback to Apple, and it definitely sounds like a bug.

Feel free to share the full FCPXML if you want us to have a proper look.

**Reply by tangierc** [reply]
Thanks. I made a timeline exclusively made of only the problematic clips and the same timeline settings as the original project, exported an XML and reimported it without issue. For some reason they're only a problem on my main project timeline, but not all lined up on their own timeline.  Here's the original project.

[Anxious-TC Cut 5-Locked Picture for Teams.fcpxmld.zip](https://github.com/CommandPost/FCPCafe/files/11683287/Anxious-TC.Cut.5-Locked.Picture.for.Teams.fcpxmld.zip)

I'll definitely send feedback to Apple.

**Reply by tangierc** [reply]
With the help of someone in the Final Cut Pro community, they told me which files were having the problem. See my list below.
I then went into the Finder, found all of those problem files, then converted them to WAV files and one Apple ProRes file.
Next I relinked those problem files in the FCP project library to these new files I created. This was successful.
I exported a new XML on the project only, without issue. Sadly FCP doesn't validate the XML on export.
When attempting to test this XML in X2Pro it came up with the same errors as my last XML.
I also could not import this new XML inside FCP.

I could use some assistance. I am attaching the new XML that still doesn't work.
![Screenshot 2023-06-08 at 6 39 49 PM](https://github.com/CommandPost/FCPCafe/assets/9577806/e1a2d2b3-de30-4115-9ab2-4b6b40604b8e)
[Anxious-TC Cut 5-Locked Picture for Teams_fixed.fcpxmld.zip](https://github.com/CommandPost/FCPCafe/files/11699418/Anxious-TC.Cut.5-Locked.Picture.for.Teams_fixed.fcpxmld.zip)

**Reply by latenitefilms** [reply]
I will have a proper look at this shortly.

**Reply by tangierc** [reply]
I did some digging on X2Pro's forums. One user who had a similar issue found that missing audio channels in the problematic files cause their XML to fail, but my clips don't have that issue. More peculiar is that these "problematic" files have do not incur these XML errors when they exist together on their own timeline with the exact same project settings as the original project.

**Reply by tangierc** [reply]
There must be something else going on. I deleted every instance of those clips and exported another XML (attached) and X2Pro still could not parse the document. 

I tried importing this XML back into FCP and now there's only one error instead of the 24 I had. This doesn't solve the issue though. I need those clips to stay in the edit, but I wonder if X2Pro is just not getting past this first XML error that always shows up:

Info.fcpxml	Invalid edit with no respective media. (/fcpxml[1]/library[1]/event[1]/project[1]/sequence[1]/spine[1]/asset-clip[17])

I'm not sure how to decipher this error.

[Anxious-TC Cut 5-Locked Picture for Teams-Test for XML.fcpxmld.zip](https://github.com/CommandPost/FCPCafe/files/11701757/Anxious-TC.Cut.5-Locked.Picture.for.Teams-Test.for.XML.fcpxmld.zip)

**Comment by latenitefilms** [comment, marked as answer]
For anyone following along, I've answered in more detail [here](https://github.com/latenitefilms/BRAWToolbox/discussions/134#discussioncomment-6141869).
